iceoryx_posh/source/roudi/memory/memory_provider.cpp

```diff
@@ -122,7 +122,7 @@ cxx::expected<MemoryProviderError> MemoryProvider::destroy() noexcept
 
     if (!destructionResult.has_error())
     {
-        rp::BaseRelativePointer::unregisterPtr(rp::BaseRelativePointer::id_t{m_segmentId});
+        rp::UntypedRelativePointer::unregisterPtr(rp::segment_id_t{m_segmentId});
```

Comment:
This is inconsistent design-wise, as elsewhere (here in `m_segmentId` we still store primitive types, which requires casting or construction later on (as here). Note that at some points we need the primitive type, to use in memcpy (trivially copyable) operations or in atomic CAS operations etc.

These low level manipulation that is required in the overall design is another reason why I consider `Newtype` to be the wrong approach here.

Reply:
See the discussion above. I would appreciate if we could discuss the `NewType` approach elsewhere e.g. in the next developer meetup and not here.

Reply:
I checked it and it's currently not possible as `NewType` can not be assigned a new value after it's creation. I would leave it as it is.

Reply:
That is also bad, as it means it is essentially `const`. This is not what I would expect from a `NewType` that (roughly) follows the functional programming idioms (e.g. from Haskell).